src/toil/jobStores/googleJobStore.py

```diff
@@ -0,0 +1,536 @@
+        else:
+            with open(sseKeyPath) as f:
+                sseKey = f.read()
+            assert len(sseKey) == 32
+            encodedSseKey = base64.b64encode(sseKey)
+            encodedSseKeyMd5 = base64.b64encode(hashlib.sha256(sseKey).digest())
+            return {'x-goog-encryption-algorithm': 'AES256',
+                    'x-goog-encryption-key': encodedSseKey,
+                    'x-goog-encryption-key-sha256': encodedSseKeyMd5,
+                    "Cache-Control": "no-store"}
+
+    def _delete(self, jobStoreID, encrypt=True):
+        headers = self.encryptedHeaders if encrypt else self.headerValues
+        try:
+            key = self._getKey(jobStoreID, headers)
+        except NoSuchFileException:
+            pass
+        else:
+            try:
+                key.delete()
+            except boto.exception.GSResponseError as e:
+                if e.status == 404:
+                    pass
+        # best effort delete associated files
+        for fileID in self.files.list(prefix=jobStoreID):
+            try:
+                self.deleteFile(fileID)
+            except NoSuchFileException:
+                pass
+
+    def _getKey(self, jobStoreID=None, headers=None):
+        # gets remote key, in contrast to self._newKey
+        key = None
+        try:
+            key = self.files.get_key(jobStoreID, headers=headers)
+        except boto.exception.GSDataError:
+            pass  # see self._writeFile for more
+        if key is None:
+            raise NoSuchFileException(jobStoreID)
+        else:
+            return key
+
+    def _newKey(self, jobStoreID):
+        # does not create a key remotely. Careful- does not ensure key name is unique
+        return self.files.new_key(jobStoreID)
+
+    def _readContents(self, jobStoreID):
+        # used on job files only, which will be encrypted if avaliable
+        headers = self.encryptedHeaders
+        return self._getKey(jobStoreID, headers).get_contents_as_string(headers=headers)
+
+    def _writeFile(self, jobStoreID, fileObj, update=False, encrypt=True):
+        headers = self.encryptedHeaders if encrypt else self.headerValues
+        if update:
+            key = self._getKey(jobStoreID=jobStoreID, headers=headers)
+        else:
+            key = self._newKey(jobStoreID=jobStoreID)
+        headers = self.encryptedHeaders if encrypt else self.headerValues
+        try:
+            key.set_contents_from_file(fileObj, headers=headers)
+        except boto.exception.GSDataError:
+            # per https://cloud.google.com/storage/docs/encryption#customer
+            # -supplied_encryption_keys the etag and md5 will not match with customer supplied
+            # keys. however boto didn't get the memo apparently, and will raise this error if
+            # they dont match reported: https://github.com/boto/boto/issues/3518
+            pass
+
+    def _writeString(self, jobStoreID, stringToUpload, update=False, encrypt=True):
```

Comment:
With wrapper methods like this, I often use **kwargs so I don't have two covariant signatures to maintain.